### ISSUE TITLE
messagePump - ClientConnectionErrors Gone Wild 

### ISSUE THREAD

**Issue by GaryOkie** [PeteRager/lennoxs30#133]
Apparently I had some sort of LAN network issue on remote vacation home that sorted itself out.   There were HASS communication errors to several other IoT devices but they were not nearly as pronounced as the comm errors from the S30 custom integration.   This is similar to issue #123 requesting improved error handling.   Not a big deal, just a lot of log spam.

```
3:50:47 PM – (ERROR) Lennox S30/E30 (custom integration) - 
message first occurred at January 6, 2022, 5:39:39 PM and shows up 256 times

MessagePump clientResponse please report this information as an Issue - Error while executing request: [Content-Length can't be present with Transfer-Encoding]: error=<class 'aiohttp.client_exceptions.ClientResponseError'>, resp.status=[400], resp.request_info=[RequestInfo(url=URL('https://192.168.0.74/Messages/hass-lake/Retrieve?Direction=Oldest-to-Newest&MessageCount=10&StartTime=1&LongPollingTimeout=15'), method='GET', headers=<CIMultiDictProxy('Host': '192.168.0.74', 'Accept': '*/*', 'Accept-Encoding': 'gzip, deflate', 'User-Agent': 'Python/3.9 aiohttp/3.8.1')>, real_url=URL('https://192.168.0.74/Messages/hass-lake/Retrieve?Direction=Oldest-to-Newest&MessageCount=10&StartTime=1&LongPollingTimeout=15'))], resp.headers=[None]

2:54:47 PM – (ERROR) /usr/local/lib/python3.9/site-packages/lennoxs30api/s30api_async.py - 
message first occurred at January 6, 2022, 5:47:36 PM and shows up 112 times


2:54:07 PM – (ERROR) Lennox S30/E30 (custom integration) - 
message first occurred at January 6, 2022, 5:39:19 PM and shows up 48 times

messagePump exception
2:54:07 PM – (ERROR) /usr/local/lib/python3.9/site-packages/lennoxs30api/s30api_async.py -
 message first occurred at January 6, 2022, 5:39:19 PM and shows up 48 times
```
```
2022-01-09 14:54:07 ERROR (MainThread) [custom_components.lennoxs30] messagePump_task - general error host [192.168.0.74] Code [5] Reference [5] [messagePump exception]
2022-01-09 14:54:09 ERROR (MainThread) [lennoxs30api.s30api_async] MessagePump clientResponse please report this information as an Issue - Error while executing request: [Content-Length can't be present with Transfer-Encoding]: error=<class 'aiohttp.client_exceptions.ClientResponseError'>, resp.status=[400], resp.request_info=[RequestInfo(url=URL('https://192.168.0.74/Messages/hass-lake/Retrieve?Direction=Oldest-to-Newest&MessageCount=10&StartTime=1&LongPollingTimeout=15'), method='GET', headers=<CIMultiDictProxy('Host': '192.168.0.74', 'Accept': '*/*', 'Accept-Encoding': 'gzip, deflate', 'User-Agent': 'Python/3.9 aiohttp/3.8.1')>, real_url=URL('https://192.168.0.74/Messages/hass-lake/Retrieve?Direction=Oldest-to-Newest&MessageCount=10&StartTime=1&LongPollingTimeout=15'))], resp.headers=[None]
2022-01-09 14:54:09 ERROR (MainThread) [custom_components.lennoxs30] messagePump_task - communication error to host [192.168.0.74] Code [11] Reference [2] [ClientResponseError Error while executing request: [Content-Length can't be present with Transfer-Encoding]: error=<class 'aiohttp.client_exceptions.ClientResponseError'>, resp.status=[400], resp.request_info=[RequestInfo(url=URL('https://192.168.0.74/Messages/hass-lake/Retrieve?Direction=Oldest-to-Newest&MessageCount=10&StartTime=1&LongPollingTimeout=15'), method='GET', headers=<CIMultiDictProxy('Host': '192.168.0.74', 'Accept': '*/*', 'Accept-Encoding': 'gzip, deflate', 'User-Agent': 'Python/3.9 aiohttp/3.8.1')>, real_url=URL('https://192.168.0.74/Messages/hass-lake/Retrieve?Direction=Oldest-to-Newest&MessageCount=10&StartTime=1&LongPollingTimeout=15'))], resp.headers=[None]]
2022-01-09 14:54:10 ERROR (MainThread) [lennoxs30api.s30api_async] MessagePump clientResponse please report this information as an Issue - Error while executing request: [Content-Length can't be present with Transfer-Encoding]: error=<class 'aiohttp.client_exceptions.ClientResponseError'>, resp.status=[400], resp.request_info=[RequestInfo(url=URL('https://192.168.0.74/Messages/hass-lake/Retrieve?Direction=Oldest-to-Newest&MessageCount=10&StartTime=1&LongPollingTimeout=15'), method='GET', headers=<CIMultiDictProxy('Host': '192.168.0.74', 'Accept': '*/*', 'Accept-Encoding': 'gzip, deflate', 'User-Agent': 'Python/3.9 aiohttp/3.8.1')>, real_url=URL('https://192.168.0.74/Messages/hass-lake/Retrieve?Direction=Oldest-to-Newest&MessageCount=10&StartTime=1&LongPollingTimeout=15'))], resp.headers=[None]
2022-01-09 14:54:10 ERROR (MainThread) [custom_components.lennoxs30] messagePump_task - communication error to host [192.168.0.74] Code [11] Reference [2] [ClientResponseError Error while executing request: [Content-Length can't be present with Transfer-Encoding]: error=<class 'aiohttp.client_exceptions.ClientResponseError'>, resp.status=[400], resp.request_info=[RequestInfo(url=URL('https://192.168.0.74/Messages/hass-lake/Retrieve?Direction=Oldest-to-Newest&MessageCount=10&StartTime=1&LongPollingTimeout=15'), method='GET', headers=<CIMultiDictProxy('Host': '192.168.0.74', 'Accept': '*/*', 'Accept-Encoding': 'gzip, deflate', 'User-Agent': 'Python/3.9 aiohttp/3.8.1')>, real_url=URL('https://192.168.0.74/Messages/hass-lake/Retrieve?Direction=Oldest-to-Newest&MessageCount=10&StartTime=1&LongPollingTimeout=15'))], resp.headers=[None]]
2022-01-09 14:54:20 ERROR (MainThread) [lennoxs30api.s30api_async] MessagePump clientResponse please report this information as an Issue - Error while executing request: [Content-Length can't be present with Transfer-Encoding]: error=<class 'aiohttp.client_exceptions.ClientResponseError'>, resp.status=[400], resp.request_info=[RequestInfo(url=URL('https://192.168.0.74/Messages/hass-lake/Retrieve?Direction=Oldest-to-Newest&MessageCount=10&StartTime=1&LongPollingTimeout=15'), method='GET', headers=<CIMultiDictProxy('Host': '192.168.0.74', 'Accept': '*/*', 'Accept-Encoding': 'gzip, deflate', 'User-Agent': 'Python/3.9 aiohttp/3.8.1')>, real_url=URL('https://192.168.0.74/Messages/hass-lake/Retrieve?Direction=Oldest-to-Newest&MessageCount=10&StartTime=1&LongPollingTimeout=15'))], resp.headers=[None]
2022-01-09 14:54:20 ERROR (MainThread) [custom_components.lennoxs30] messagePump_task - communication error to host [192.168.0.74] Code [11] Reference [2] [ClientResponseError Error while executing request: [Content-Length can't be present with Transfer-Encoding]: error=<class 'aiohttp.client_exceptions.ClientResponseError'>, resp.status=[400], resp.request_info=[RequestInfo(url=URL('https://192.168.0.74/Messages/hass-lake/Retrieve?Direction=Oldest-to-Newest&MessageCount=10&StartTime=1&LongPollingTimeout=15'), method='GET', headers=<CIMultiDictProxy('Host': '192.168.0.74', 'Accept': '*/*', 'Accept-Encoding': 'gzip, deflate', 'User-Agent': 'Python/3.9 aiohttp/3.8.1')>, real_url=URL('https://192.168.0.74/Messages/hass-lake/Retrieve?Direction=Oldest-to-Newest&MessageCount=10&StartTime=1&LongPollingTimeout=15'))], resp.headers=[None]]
2022-01-09 14:54:39 ERROR (MainThread) [lennoxs30api.s30api_async] MessagePump clientResponse please report this information as an Issue - Error while executing request: [Content-Length can't be present with Transfer-Encoding]: error=<class 'aiohttp.client_exceptions.ClientResponseError'>, resp.status=[400], resp.request_info=[RequestInfo(url=URL('https://192.168.0.74/Messages/hass-lake/Retrieve?Direction=Oldest-to-Newest&MessageCount=10&StartTime=1&LongPollingTimeout=15'), method='GET', headers=<CIMultiDictProxy('Host': '192.168.0.74', 'Accept': '*/*', 'Accept-Encoding': 'gzip, deflate', 'User-Agent': 'Python/3.9 aiohttp/3.8.1')>, real_url=URL('https://192.168.0.74/Messages/hass-lake/Retrieve?Direction=Oldest-to-Newest&MessageCount=10&StartTime=1&LongPollingTimeout=15'))], resp.headers=[None]
2022-01-09 14:54:39 ERROR (MainThread) [custom_components.lennoxs30] messagePump_task - communication error to host [192.168.0.74] Code [11] Reference [2] [ClientResponseError Error while executing request: [Content-Length can't be present with Transfer-Encoding]: error=<class 'aiohttp.client_exceptions.ClientResponseError'>, resp.status=[400], resp.request_info=[RequestInfo(url=URL('https://192.168.0.74/Messages/hass-lake/Retrieve?Direction=Oldest-to-Newest&MessageCount=10&StartTime=1&LongPollingTimeout=15'), method='GET', headers=<CIMultiDictProxy('Host': '192.168.0.74', 'Accept': '*/*', 'Accept-Encoding': 'gzip, deflate', 'User-Agent': 'Python/3.9 aiohttp/3.8.1')>, real_url=URL('https://192.168.0.74/Messages/hass-lake/Retrieve?Direction=Oldest-to-Newest&MessageCount=10&StartTime=1&LongPollingTimeout=15'))], resp.headers=[None]]
2022-01-09 14:54:47 ERROR (MainThread) [lennoxs30api.s30api_async] MessagePump clientResponse please report this information as an Issue - Error while executing request: [Content-Length can't be present with Transfer-Encoding]: error=<class 'aiohttp.client_exceptions.ClientResponseError'>, resp.status=[400], resp.request_info=[RequestInfo(url=URL('https://192.168.0.74/Messages/hass-lake/Retrieve?Direction=Oldest-to-Newest&MessageCount=10&StartTime=1&LongPollingTimeout=15'), method='GET', headers=<CIMultiDictProxy('Host': '192.168.0.74', 'Accept': '*/*', 'Accept-Encoding': 'gzip, deflate', 'User-Agent': 'Python/3.9 aiohttp/3.8.1')>, real_url=URL('https://192.168.0.74/Messages/hass-lake/Retrieve?Direction=Oldest-to-Newest&MessageCount=10&StartTime=1&LongPollingTimeout=15'))], resp.headers=[None]
2022-01-09 14:54:47 ERROR (MainThread) [custom_components.lennoxs30] messagePump_task - communication error to host [192.168.0.74] Code [11] Reference [2] [ClientResponseError Error while executing request: [Content-Length can't be present with Transfer-Encoding]: error=<class 'aiohttp.client_exceptions.ClientResponseError'>, resp.status=[400], resp.request_info=[RequestInfo(url=URL('https://192.168.0.74/Messages/hass-lake/Retrieve?Direction=Oldest-to-Newest&MessageCount=10&StartTime=1&LongPollingTimeout=15'), method='GET', headers=<CIMultiDictProxy('Host': '192.168.0.74', 'Accept': '*/*', 'Accept-Encoding': 'gzip, deflate', 'User-Agent': 'Python/3.9 aiohttp/3.8.1')>, real_url=URL('https://192.168.0.74/Messages/hass-lake/Retrieve?Direction=Oldest-to-Newest&MessageCount=10&StartTime=1&LongPollingTimeout=15'))], resp.headers=[None]]
2022-01-09 14:58:54 ERROR (MainThread) [custom_components.lennoxs30] messagePump_task - communication error to host [192.168.0.74] Code [11] Reference [4] [messagePump - Server Disconnected]
2022-01-09 15:03:20 ERROR (MainThread) [custom_components.lennoxs30] messagePump_task - communication error to host [192.168.0.74] Code [11] Reference [4] [messagePump - Server Disconnected]
2022-01-09 15:10:46 ERROR (MainThread) [custom_components.lennoxs30] messagePump_task - communication error to host [192.168.0.74] Code [11] Reference [4] [messagePump - Server Disconnected]
2022-01-09 15:28:25 ERROR (MainThread) [custom_components.lennoxs30] messagePump_task - communication error to host [192.168.0.74] Code [11] Reference [4] [messagePump - Server Disconnected]
2022-01-09 15:37:07 ERROR (MainThread) [custom_components.lennoxs30] messagePump_task - communication error to host [192.168.0.74] Code [11] Reference [4] [messagePump - Server Disconnected]
```

**Comment by PeteRager** [comment]
Thanks for reporting.  There’s some diagnostic info we’ve been attempting to capture on some intermittent errors so this should help that. And yes, I agree it should log a bit less!

**Comment by GaryOkie** [comment]
Just updated to  0.2.2/ 0.1.5 and checked logs to make sure all was well.    I was seeing a lot of the same errors as reported above indicating a communication error
`2022-01-31 08:31:05 ERROR (MainThread) [custom_components.lennoxs30] messagePump_task - communication error to host [192.168.0.74] Code [11] Reference [2] [ClientResponseError Error while executing request: [Content-Length can't be present with Transfer-Encoding]: error=<class 'aiohttp.client_exceptions.ClientResponseError'>, resp.status=[400], resp.request_info=[RequestInfo(url=URL('https://192.168.0.74/Messages/hass-lake/Retrieve?Direction=Oldest-to-Newest&MessageCount=10&StartTime=1&LongPollingTimeout=15'), method='GET', headers=<CIMultiDictProxy('Host': '192.168.0.74', 'Accept': '*/*', 'Accept-Encoding': 'gzip, deflate', 'User-Agent': 'Python/3.9 aiohttp/3.8.1')>, real_url=URL('https://192.168.0.74/Messages/hass-lake/Retrieve?Direction=Oldest-to-Newest&MessageCount=10&StartTime=1&LongPollingTimeout=15'))], resp.headers=[None]]`

I then pinged the device and confirmed that there was indeed a comm/network issue that was causing a intermitten high latency and packet loss...
```
--- 192.168.0.74 ping statistics ---
28 packets transmitted, 16 packets received, 42% packet loss
round-trip min/avg/max = 2.777/910.956/2077.017 ms

--- 192.168.0.74 ping statistics ---
37 packets transmitted, 8 packets received, 78% packet loss
round-trip min/avg/max = 10.243/1000.740/2001.655 ms
```
No other devices on the internal network were experiencing comm errors - just the S30.   I was just about to reboot the router then I noticed a new message in the log...
`2022-01-31 08:31:08 WARNING (MainThread) [lennoxs30api.s30api_async] S30 has rebooted sysId [LCC] old uptime [1903320] new uptime [60]`

Huh, the S30 apparently detected a problem and spontaneously rebooted?!   Indeed, all the comm errors cleared up and 0 packet loss now.  I assume the lennoxs30api did not force the reboot.    I'm wondering now if I need to contact service to have the S30 replaced?

**Comment by PeteRager** [comment]
Yes, I believe the S30 has some sort of built in health check / hardware watchdog - so that If the software crashes / runs out of memory, etc. in the S30 - it will reboot automatically and go back to controlling the HVAC system.  I believe the crash logs get delivered to Lennox via an api and hopefully they review and fix stuff.  There fix may be to turn that local port off.  I’ve disabled automatic updates on my S30, as I don’t want to lose local control.   

We put that warning in to detect when it does this as we though it was related to enabling diagnostic mode to get the power inverter data.

The ping issues could be a network problem or it could have been the S30 getting hosed up.

Good information.  These also seem to manifest on restarts.

**Comment by PeteRager** [comment]
There is some diagnostic data that the S30 sends regarding the WiFi connection.  I have seen complaints on the Lennox forum about misbehavior when the WiFi connection is not strong enough.  I don't know your setup.  The Wifi Connection is to the box near the HVAC system (mine has an antennae on it) not to the panel.

If you enable message logging to a file.  Periodically they'll be messages output like this.  I believe the rssi is the signal strength.  Mine looks to be -60; which according to this is OK - https://www.metageek.com/training/resources/understanding-rssi/

My wifi is in the basement and the S30 in the sttic.  I'm in a quiet Wifi location (no neighboring houses. etc.)

I'll look at putting this info into a sensor as I work on handling these connection errors better.  In the meantime, if you have the disk space consider enabling message logging, to look at the rssi over time and maybe we'll see other diagnostic info from the S30.


```
    "messages": [
        {
            "MessageId": 0,
            "SenderID": "**redacted**",
            "TargetID": "ha_prod",
            "MessageType": "PropertyChange",
            "Data": {
                "interfaces": [
                    {
                        "publisher": {
                            "publisherName": "lcc"
                        },
                        "Info": {
                            "status": {
                                "macAddr": "**redacted**",
                                "ssid": "**redacted**",
                                "ip": "**redacted**",
                                "router": "**redacted**",
                                "networkStatus": "enabled",
                                "channel": 3,
                                "dns": "**redacted**",
                                "dns2": "",
                                "subnetMask": "**redacted**",
                                "bitRate": 72200000,
                                "speed": " 72.2 MBit per second",
                                "rssi": -60
                            },
                            "APDetails": {
                                "rssi": -60,
                                "security": "WPA-Personal",
                                "password": "**redacted**",
                                "ssid": "**redacted**",
                                "bssid": "58:6d:8f:ed:ea:be"
                            },
                            "diagnostics": {
                                "ssid": "**redacted**",
                                "bssid": "58:6d:8f:ed:ea:be",
                                "txByteCount": 10775785,
                                "rxByteCount": 0,
                                "ip4addr": "**redacted**",
                                "txPacketCount": 38219,
                                "rxPacketCount": 0
                            }
                        },
                        "id": 0
                    }
                ]
            }
        }
    ]
```

**Comment by GaryOkie** [comment]
Thanks Pete for the help.    I had  "Debug logging logs messages" already enabled, but have now just enabled "Log S30 message to seperate file"  and entered just "S30.log".    Not sure if that will write to the root or to /config - but no matter, I have plenty of SSD space.

This S30 is at a vacation home with little to no neighboring WiFi interference.    The router is on the same floor as the iComfort hub just 15' away on a dedicated  2.4Ghz SSID with only 5 wireless devices.  I expect the signal strength to be quite good.    

I'll get back to you as soon as I get a substantial amount of rssi data over the next 24 hours.

**Comment by GaryOkie** [comment]
Debug running for nearly 8 hours now - log contains 660 instances of "rssi" - and all are -25, supposedly proving no variation in WiFi signal.  

No lennoxs30api errors in the HASS log for over 6 hours, and then at 14:55 the comm errors started.  ping was showing very high latency and major packet loss. No watchdog reboot has occurred yet.   

I just pinged all the other WiFi devices on this same SSID and all are responding very quickly, so there is no general network issue.

The S30 debug log does not show any errors that I saw.  It just shows gaps in the log when the comm errors are occurring.

At the moment - the S30 is not responding to the integration, to the mobile app, or to the iComfort portal.  Worst possible time for it to be acting up.   Severe cold snap approaching tonight.  Expecting a low of 4F and ice buildup.

Sure hope the TStat manages to keep the heat on independently of whatever this comm error is.   I do have other temperature sensors around the house that are monitored.

**Comment by PeteRager** [comment]
That does not sound like a good situation.  I do not know how to perform a remote reset of the S30, though if we can’t talk to it wouldn’t help anyway.   There have been no changes to the comms layer in 0.2.2 - we are just grabbing a couple additional fields out of the messages related to smart away.  But, the situation has gotten worse for you since that upgrade.   

You may want to try disabling the integration and let it sit for a bit?

Sounds like you are able to VPN into that network.  The iOS phone app uses a direct connection to the S30 that is different than the path the integration uses.  If you can get it on that LAN, that is worth a try.

Any special firewalls, dns, VLANs that could prevent the S30 from communicating to the cloud?

What is the S30 firmware version?  That info is in the s30 state entity 

What do you have app_id set to?  I have seen changing this can sometimes fix issues.  Try something like ha_prod

We finally got a warmup here (upstate NY) been killer cold.  -20 the other night.  I get the stress.

**Comment by PeteRager** [comment]
When you have time, please post the S30 integration settings (screen shots or the entry from .storage/core.config_entries)

**Comment by GaryOkie** [comment]
Pete,  I seriously doubt this comm problem has anything to do with the latest S30 update, but I will disable it overnight.    Thanks so much for taking the time to review the issue and offering suggestions to try.

My S30 is also 3.81.207 and the hub is 3.81.777.  The app_id is HASS-Lake, and I just changed it to HASS-Lake2 and restarted.  (didn't help)

The S30 is able to connect to the cloud.  I can log into the iComfort portal and it initially shows offline, then briefly online with the temp gauge, then seconds later offline again.   This sort of matches what I see with the pings that extremely slow or not responding at all.  

```
PING 192.168.0.74 (192.168.0.74): 56 data bytes
64 bytes from 192.168.0.74: seq=187 ttl=64 time=2072.262 ms
64 bytes from 192.168.0.74: seq=188 ttl=64 time=1072.217 ms
64 bytes from 192.168.0.74: seq=189 ttl=64 time=72.141 ms
64 bytes from 192.168.0.74: seq=190 ttl=64 time=204.287 ms
64 bytes from 192.168.0.74: seq=191 ttl=64 time=57.101 ms
64 bytes from 192.168.0.74: seq=192 ttl=64 time=336.933 ms

808 packets transmitted, 6 packets received, 99% packet loss
round-trip min/avg/max = 57.101/635.823/2072.262 ms
```

Config:
```
"entry_id": "82dff3c16f29d2668e3d890704deea18",
                "version": 1,                         
                "domain": "lennoxs30",                
                "title": "Main Floor T-Stat",         
                "data": {                                      
                    "app_id": "hass-lake2",           
                    "create_sensors": true,           
                    "allergen_defender_switch": true, 
                    "create_inverter_power": true,    
                    "scan_interval": 1,               
                    "init_wait_time": 30,                                                               
                    "fast_scan_interval": 0.75,               
                    "protocol": "https",              
                    "pii_in_message_logs": false,     
                    "message_debug_logging": true,                                                      
                    "log_messages_to_file": true.                                                      
                    "message_debug_file": "S30.log",                             
                    "host": "192.168.0.74",           
                    "cloud_connection": false         
                },                                    
                "options": {},                        
                "pref_disable_new_entities": false,   
                "pref_disable_polling": false,        
                "source": "user",                     
                "unique_id": "lennoxs30_192.168.0.74",
                "disabled_by": null
```

**Comment by PeteRager** [comment]
Glad it appears to be working and heating your house.

it does appear that the S30 is having problems.  As with any electronic gadget power cycling it would be a good idea.  It’s software reboot probably doesn’t do this and it looks like it’s a WiFi radio / chip set issue.

I see you have inverter power enabled, did you also set the diagnostic level on the S30 to obtain the power inverter data?  Thereis a number entity to do this.  If so I’d try to reset it to 0.

**Comment by GaryOkie** [comment]
Yes, as long as the S30 is still functioning as a normal T-Stat and operating the Heat Pump at the last temp I had set (60F)  - I'm not too concerned about it for now.   I really had wanted to monitor the emergency heat and all.   

Yes, I had created a number switch to enable the inverter.   It was set to "1".   I never used 2.    

Interesting - the S30 is responding to pings now quickly.   Working as it should from the iComfort app.   I haven't re-enabled the integration yet.    I'll go ahead and re-enable it, but keep the inverter info off for now.

**Comment by PeteRager** [comment]
Yes, I would try restarting the integration and return diagnostic level to 0 as soon as possible and leaving it like that an assess stability.

**Comment by GaryOkie** [comment]
I've done exactly that.    All is well for now.   I can monitor whole-house energy usage so will be able to derive emergency electrical heat spikes without the inverter-specific data.  I had forgotten this inverter option was an "alpha" feature.    So there is a chance that was affecting my connectivity to the S30 somehow (at level 1, not 2)?

**Comment by GaryOkie** [comment]
Now seeing a new error...
```
2022-02-02 09:31:56 ERROR (MainThread) [lennoxs30api.s30api_async] executeOnUpdateCallback - failed
Traceback (most recent call last):
File "/usr/local/lib/python3.9/site-packages/lennoxs30api/s30api_async.py", line 992, in executeOnUpdateCallbacks
callbackfunc()
File "/config/custom_components/lennoxs30/sensor.py", line 92, in update_callback
self.schedule_update_ha_state()
File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 643, in schedule_update_ha_state
self.hass.add_job(self.async_update_ha_state(force_refresh)) # type: ignore
AttributeError: 'NoneType' object has no attribute 'add_job'
2022-02-02 09:33:56 ERROR (MainThread) [lennoxs30api.s30api_async] executeOnUpdateCallback - failed
Traceback (most recent call last):
File "/usr/local/lib/python3.9/site-packages/lennoxs30api/s30api_async.py", line 992, in executeOnUpdateCallbacks
callbackfunc()
File "/config/custom_components/lennoxs30/sensor.py", line 92, in update_callback
self.schedule_update_ha_state()
File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 643, in schedule_update_ha_state
self.hass.add_job(self.async_update_ha_state(force_refresh)) # type: ignore
AttributeError: 'NoneType' object has no attribute 'add_job'
2022-02-02 09:34:57 ERROR (MainThread) [lennoxs30api.s30api_async] executeOnUpdateCallback - failed
Traceback (most recent call last):
File "/usr/local/lib/python3.9/site-packages/lennoxs30api/s30api_async.py", line 992, in executeOnUpdateCallbacks
callbackfunc()
File "/config/custom_components/lennoxs30/sensor.py", line 92, in update_callback
self.schedule_update_ha_state()
File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 643, in schedule_update_ha_state
self.hass.add_job(self.async_update_ha_state(force_refresh)) # type: ignore
AttributeError: 'NoneType' object has no attribute 'add_job'
2022-02-02 09:36:55 ERROR (MainThread) [lennoxs30api.s30api_async] executeOnUpdateCallback - failed
Traceback (most recent call last):
File "/usr/local/lib/python3.9/site-packages/lennoxs30api/s30api_async.py", line 992, in executeOnUpdateCallbacks
callbackfunc()
File "/config/custom_components/lennoxs30/sensor.py", line 92, in update_callback
self.schedule_update_ha_state()
File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 643, in schedule_update_ha_state
self.hass.add_job(self.async_update_ha_state(force_refresh)) # type: ignore
AttributeError: 'NoneType' object has no attribute 'add_job'
```
Re-enabling the integration was not as straightforward as it should have been as all the entities were still showing "disabled by device" after the S30 integration was enabled.   Had to manually enable the main climate entity and tried to enable the other sensors/switches but they were stuck disabled.    This may be the cause of the above errors.

 I think the best thing now is to delete it and reinstall fresh.

**Comment by PeteRager** [comment]
Yes, that sounds like an issue.  Looks like that path needs more testing by me 😣

yes, I think delete and reinstall is the right approach.   We’ll want the power inverter sensor enabled so we get the diagnostic level number.  Once it is up set to zero.

**Comment by GaryOkie** [comment]
update - reinstalled S30 integration from scratch - that went without a hitch.

Exactly 11 minutes after the HASS restart, connection errors started again....

```
2022-02-02 11:30:44 WARNING (SyncWorker_3) [homeassistant.loader] We found a custom integration lennoxs30 which has not been tested by Home Assistant. This component might cause stability problems, be sure to disable it if you experience issues with Home Assistant


2022-02-02 11:41:44 ERROR (MainThread) [custom_components.lennoxs30] messagePump_task - communication error to host [192.168.0.74] Code [11] Reference [4] [messagePump - Server Disconnected]
2022-02-02 11:41:45 ERROR (MainThread) [custom_components.lennoxs30] messagePump_task - communication error to host [192.168.0.74] Code [11] Reference [3] [messagePump - ClientConnectionError Cannot connect to host 192.168.0.74:443 ssl:False [Connect call failed ('192.168.0.74', 443)]]
2022-02-02 11:41:46 ERROR (MainThread) [custom_components.lennoxs30] messagePump_task - communication error to host [192.168.0.74] Code [11] Reference [3] [messagePump - ClientConnectionError Cannot connect to host 192.168.0.74:443 ssl:False [Connect call failed ('192.168.0.74', 443)]]
2022-02-02 11:41:47 ERROR (MainThread) [custom_components.lennoxs30] messagePump_task - communication error to host [192.168.0.74] Code [11] Reference [3] [messagePump - ClientConnectionError Cannot connect to host 192.168.0.74:443 ssl:False [Connect call failed ('192.168.0.74', 443)]]
2022-02-02 11:41:48 ERROR (MainThread) [custom_components.lennoxs30] messagePump_task - communication error to host [192.168.0.74] Code [11] Reference [3] [messagePump - ClientConnectionError Cannot connect to host 192.168.0.74:443 ssl:False [Connect call failed ('192.168.0.74', 443)]]
2022-02-02 11:41:49 ERROR (MainThread) [custom_components.lennoxs30] messagePump_task - communication error to host [192.168.0.74] Code [11] Reference [3] [messagePump - ClientConnectionError Cannot connect to host 192.168.0.74:443 ssl:False [Connect call failed ('192.168.0.74', 443)]]
2022-02-02 11:41:51 ERROR (MainThread) [custom_components.lennoxs30] reinitialize_task host [192.168.0.74] Code [11] Reference [3] [login failed due host not reachable url [https://192.168.0.74/Endpoints/HASS-Lake/Connect] Cannot connect to host 192.168.0.74:443 ssl:False [Connect call failed ('192.168.0.74', 443)]]
```

I found the inverter number switch was reset to "1" after the restart so I changed this to 0, and haven't seen the connection errors for the past 15 minutes.   If this holds up, it likely implicates the S30 integration inverter sensor triggering a connection problem.   And after several hours of this - the S30 eventually goes haywire/offline?


BTW - the inverter sensor is still showing a value of 1,967 Watts which must have been the last reading.   Obviously this value needs to be ignored once the inverter/number switch is zero.

**Comment by PeteRager** [comment]
Good, sounds like we may be getting back to a point of stability.  Could just be luck..

Let’s run it in diagnostic mode zero for a while (week or so) which will give us a good assessment.  

At level 0, the data needed by that sensor does not get delivered hence it will hold that value.  So yeah, it’s not going to be very useful right now.

I had observed instability on my S30 when using diagnostic mode.  The code was a contribution and appears to be working fine on his system.  I believe his system does not have a Lennox account and hence does not send data to the cloud.  Hence I’ve marked it as alpha until I get feedback that it is stable in more than one system.

I believe that every client has an independent subscription - e.g. the integration, each mobile app,… Whenever data changes it is sent / queued to each client.  Clients do not always disconnect well, so for example if the Mobile app is closed or the integration is stopped, messages for these clients remain in the queue for some amount of time.

When diagnostic mode is enabled much more data is generated and in your system the production of data may be exceeding the consumption and then we see this cascading set of failures.

Lennox has a diagnostic app that runs on tablets. The way it is supposed to be used is this.  Technician shows up at your house, they press a button on the S30 controller that enables a temporary WiFi network (and hence disconnects from your wifi.). The app connects to the 443 port, send command for diag level 2 and then the data streams into his tablet for display.

**Comment by PeteRager** [comment]
@GaryOkie - has the system remained stable since the change to diagnostic level?

**Comment by GaryOkie** [comment]
sounds great - thanks for reducing the error messages, especially those that trigger python tracebacks upon a timeout.  

I don't have my S30 blocked from the Internet at this time, but I do have it configured for  local only mode.

**Comment by PeteRager** [comment]
How do you configure it for local only mode?

If you want to try to get that power inverter working, I think we can say diag level 1 is problematic.  You could try diag level 2 or not…

**Comment by GaryOkie** [comment]
I just set up the integration using an IP address and did not provide a Lennox ID/password.   That's what I meant by "local only" from a HASS/S30 integration perspective.   Of course, since it is not blocked from the Internet, the S30 will phone home to provide stats and for remote portal and app access, so in that sense, it sure isn't local only.   Sorry I wasn't clearer.

This reminds me - during troubleshooting I wanted to try a different subnet, but it sure wasn't obvious how to change the IP without reinstalling from scratch.  I did find a way by editing the .storage core entity file, but shouldn't the IP address (and optional user/password) be part of the config flow UI settings?

**Comment by PeteRager** [comment]
Yeah, I think we can allow that.  Let’s add an issue.  I edit the storage file, no issues with doing that.

**Comment by GaryOkie** [comment]
That would be helpful updating the config UI, as editing .storage files is similar to editing MS registry.  No issue if you know where to look!

I'll leave my system at diag 0.   I need stability for now while away.

**Comment by PeteRager** [comment]
This issue should be fully resolved in the  0.2.4 release; which will be released by tomorrow.
